### PR TITLE
Null safe and unsafe in a single codebase

### PR DESCRIPTION
**Pull request by domesticmouse**
Crunching both null safety and null unsafe into the same codebase to make the Cloud Run publishing cycle easier.

/cc @johnpryan this is the context for https://github.com/dart-lang/dart-pad/pull/1812 =)

### CODE REVIEW

File: lib/src/compiler.dart [hunk 1]
```diff
@@ -150,12 +158,19 @@ class Compiler {
         '--modules=amd',
         if (usingFlutter) ...[
           '-s',
-          _flutterWebManager.summaryFilePath,
+          _flutterWebManager.summaryFilePath(_nullSafety),
           '-s',
-          '${Sdk.flutterBinPath}/cache/flutter_web_sdk/flutter_web_sdk/kernel/flutter_ddc_sdk.dill'
+          if (_nullSafety)
+            '${Sdk.flutterBinPath}/cache/flutter_web_sdk/flutter_web_sdk/kernel/flutter_ddc_sdk_sound.dill'
```
Comment (srawlins) [inline lib/src/compiler.dart:164]: Perhaps worth adding a field, `Sdk.flutterWebSdkKernelPath` or similar?

**Review by srawlins — APPROVED** [review]
Very clean. I only spot-reviewed `common.dart`. I like all of the grind changes.

File: lib/src/compiler.dart (hunk 1 above)
Comment (domesticmouse) [reply lib/src/compiler.dart:164]: I'm curious if there will be usages apart from here for that method?

Comment (srawlins) [reply lib/src/compiler.dart:164]: Fair. The line is just so long 😛 .

Comment (domesticmouse) [reply lib/src/compiler.dart:164]: true dat =)

**Review by RedBrogdon — APPROVED** [review]
Is it weird that the most interesting part of this is the grinder file? 😄

File: lib/services_dev.dart [hunk 2]
```diff
@@ -63,21 +62,19 @@ class EndpointsServer {
...
-  final String proxyTarget;
 
   Pipeline pipeline;
   Handler handler;
 
   CommonServerApi commonServerApi;
   FlutterWebManager flutterWebManager;
 
-  EndpointsServer._(this.port, this.proxyTarget) {
-    final commonServerImpl = (proxyTarget != null && proxyTarget.isNotEmpty)
```
[the remark this answers is not in the document]
Comment (domesticmouse) [reply lib/services_dev.dart:75]: https://github.com/dart-lang/dart-services/pull/691 =)

File: lib/src/compiler.dart (hunk 1 above)
Comment (domesticmouse) [reply lib/src/compiler.dart]: I'm not quite sure what a concurrent literal is, but how about a ternary operator to crunch out the common substring?

**Pull request merged**